common/dbconnector.cpp

```
@@ -128,10 +134,16 @@ void SonicDBConfig::initializeGlobalConfig(const string &file)
                     continue;
                 }
 
-                parseDatabaseConfig(local_file, inst_entry, db_entry, separator_entry);
-                m_inst_info[key] = inst_entry;
-                m_db_info[key] = db_entry;
-                m_db_separator[key] = separator_entry;
+                parseDatabaseConfig(local_file, inst_entry, db_entry, separator_entry, true);
+                // all the entries are empty, then don't add them to the map. It may happen if the included
+                // config doesn't exist. For example, dash-ha container will only mount the redis instance
```

>dash-ha container will only mount the redis instance [](http://example.com/codeflow?start=54&length=52)

This should be considered as a config file issue instead of implementation issue. We'd better not to assume "The global config file includes all possible includes in the switch."

Thw swss-common deb package just provide a sample database_config.json which may be useful for most of the use cases, however the installer environment need to ensure the real database config files valid, and we should not relax the implementation on parsing errors.

---

This is not about database_config.json but database_global.json. The latter includes multiple  database_config.json and not all of them are accessible to the container. This does not assume global config includes all database_config. It works if global config only include a subset of all possible includes. 

The PR only ignores the error if the included file is not found for container case. It provides INFO log for troubleshooting, which should be sufficient  if it is a legitimate config issue.

Alternatively, we can have multiple database_global.json, one per container. And we need a naming scheme for containers to find their corresponding database_global.json. This may require a bit of redesign in database.

@Pterosaur , please share your thoughts.

---

got it, thanks!